agent/engine/host_resource_manager.go

```diff
@@ -67,13 +67,25 @@ func (e *ResourceIsNilForTask) Error() string {
 	return fmt.Sprintf("resource %s is nil in task resources", e.resource)
 }
 
+func toStringSlice(s []*string) []string {
+	var t []string
+	for _, ptr := range s {
+		if ptr == nil {
+			t = append(t, "nil")
+		} else {
+			t = append(t, *ptr)
+		}
+	}
+	return t
+}
+
 func (h *HostResourceManager) logResources(msg string, taskArn string) {
 	logger.Debug(msg, logger.Fields{
 		"taskArn":   taskArn,
 		"CPU":       *h.consumedResource[CPU].IntegerValue,
 		"MEMORY":    *h.consumedResource[MEMORY].IntegerValue,
-		"PORTS_TCP": h.consumedResource[PORTSTCP].StringSetValue,
-		"PORTS_UDP": h.consumedResource[PORTSUDP].StringSetValue,
+		"PORTS_TCP": toStringSlice(h.consumedResource[PORTSTCP].StringSetValue),
```

Comment:
I think you could use `aws.StringValueSlice()` to accomplish this

https://github.com/aws/aws-sdk-go/blob/main/aws/convert_types.go#L31

Reply:
Thanks! This is what I need.